crates/walrus-service/src/client/cli/cli_output.rs

```diff
@@ -289,6 +310,95 @@ impl CliOutput for DryRunOutput {
     }
 }
 
+/// Get the stored quilt patches from a quilt index.
+fn get_stored_quilt_patches(
+    quilt_index: &QuiltIndex,
+    quilt_blob_id: BlobId,
+) -> Vec<StoredQuiltPatch> {
+    quilt_index
+        .patches()
+        .iter()
+        .map(|patch| {
+            StoredQuiltPatch::new(
+                quilt_blob_id,
+                &patch.identifier,
+                patch.quilt_patch_internal_id(),
+            )
+        })
+        .collect()
+}
+
+impl CliOutput for StoreQuiltDryRunOutput {
+    fn print_cli_output(&self) {
+        self.quilt_blob_output.print_cli_output();
+        construct_stored_quilt_patch_table(&get_stored_quilt_patches(
+            &self.quilt_index,
+            self.quilt_blob_output.blob_id,
+        ))
+        .printstd();
+    }
+}
+
+impl CliOutput for QuiltIndex {
+    fn print_cli_output(&self) {
+        match self {
+            QuiltIndex::V1(indexv1) => indexv1.print_cli_output(),
+        }
+    }
+}
+
+impl CliOutput for QuiltIndexV1 {
+    fn print_cli_output(&self) {
+        println!(
+            "{}",
+            format!(
+                "Quilt Index V1, total number of patches: {}",
+                self.quilt_patches.len()
+            )
+            .bold()
+            .walrus_purple()
+        );
+
+        let mut table = Table::new();
+        table.set_format(default_table_format());
+        table.set_titles(row![
+            b->"Index",
+            b->"Index Range",
+            b->"Identifier"
+        ]);
+
+        for (i, quilt_patch) in self.patches().iter().enumerate() {
+            table.add_row(row![
+                bFc->format!("{i}"),
+                format!("{}..{}", quilt_patch.start_index, quilt_patch.end_index),
+                quilt_patch.identifier
+            ]);
+        }
```

Comment:
Question/suggestion: Could we just use (an adapted version of?) the `construct_stored_quilt_patch_table` function here?